### ISSUE TITLE
Another crash when using a prepared statement from a subclass

### ISSUE THREAD

**Issue by theodorejb** [Azure/msphpsql#78]
I tested the just-released x64 driver on PHP 7.0.4, and while I didn't experience a crash when running the exact script in issue #74, I found that tweaking the script slightly still causes the driver to crash. All I changed was to store the connection resource in a class and pass a parameter to the `Statement` parent constructor.

Before running the script, create a `Users` table like this:

```sql
CREATE TABLE Users (
    user_id int IDENTITY PRIMARY KEY,
    name varchar(50) NOT NULL
);

INSERT INTO Users (name)
VALUES ('Martin'), ('Emmett');
```

Then try running this:

```php
$connection = sqlsrv_connect('Computer-Name\SQLEXPRESS', ['Database' => 'MyDatabase']);
$db = new SQL($connection);

$sql = "UPDATE Users SET name = ? WHERE user_id = ?";
$id = $name = null;
$stmt = $db->prepare($sql, [&$name, &$id]);

$names = [
    1 => 'Michael J. Fox',
    2 => 'Christopher A. Lloyd',
];

foreach ($names as $id => $name) {
    $stmt->execute();
}

$stmt->close();

class SQL
{
    private $connection;

    public function __construct($connection)
    {
        $this->connection = $connection;
    }

    public function prepare($sql, array $params)
    {
        if (!$stmt = sqlsrv_prepare($this->connection, $sql, $params)) {
            throw new Exception('Failed to prepare query: ' . print_r(sqlsrv_errors(), true));
        }

        return new Statement($stmt, true);
    }
}

class BaseStatement
{
    protected $usedPrepare;

    public function __construct($usedPrepare)
    {
        $this->usedPrepare = $usedPrepare;
    }
}

class Statement extends BaseStatement
{
    private $stmt;

    public function __construct($stmt, $usedPrepare)
    {
        parent::__construct($usedPrepare); // crash occurs unless I comment out this line
        $this->stmt = $stmt;
    }

    public function execute()
    {
        sqlsrv_execute($this->stmt);
    }

    public function close()
    {
        sqlsrv_free_stmt($this->stmt);
    }
}
```

The script crashes unless I comment out the parent constructor call.

![8cf5d1c8-c932-11e5-8870-046e539f205d](https://cloud.githubusercontent.com/assets/3053271/13768337/d807cc4e-ea42-11e5-9540-e9c5b77105e8.PNG)

**Comment by meet-bhagdev** [comment]
Thanks for the quick feedback. We will look into this and share our findings with you. Did this release help solve #74?

**Comment by theodorejb** [comment]
Yes, the specific script in #74 is no longer crashing for me.

**Comment by meet-bhagdev** [comment]
@theodorejb we have attempted to fix this issue with the v4.0.3 release, can you help us validate?

**Comment by theodorejb** [comment]
:+1: This does appear to be fixed now. Thanks!
